stock_request/models/stock_move.py

```diff
@@ -55,24 +55,82 @@ def copy_data(self, default=None):
         if "allocation_ids" not in default:
             default["allocation_ids"] = []
         for alloc in self.allocation_ids:
-            default["allocation_ids"].append(
-                (
-                    0,
-                    0,
-                    {
-                        "stock_request_id": alloc.stock_request_id.id,
-                        "requested_product_uom_qty": alloc.requested_product_uom_qty,
-                    },
+            if alloc.open_product_qty:
+                default["allocation_ids"].append(
+                    (
+                        0,
+                        0,
+                        {
+                            "stock_request_id": alloc.stock_request_id.id,
+                            "requested_product_uom_qty": alloc.open_product_qty,
+                        },
+                    )
                 )
-            )
         return super(StockMove, self).copy_data(default)
 
     def _action_cancel(self):
         res = super()._action_cancel()
         self.mapped("allocation_ids.stock_request_id").check_done()
         return res
 
+    @api.model
+    def _stock_request_confirm_done_message_content(self, message_data):
+        title = _("Receipt confirmation %s for your Request %s") % (
+            message_data["picking_name"],
+            message_data["request_name"],
+        )
+        message = "<h3>%s</h3>" % title
+        message += _(
+            "The following requested items from Stock Request %s "
+            "have now been received in %s using Picking %s:"
+        ) % (
+            message_data["request_name"],
+            message_data["location_name"],
+            message_data["picking_name"],
+        )
+        message += "<ul>"
+        message += _("<li><b>%s</b>: Transferred quantity %s %s</li>") % (
+            message_data["product_name"],
+            message_data["product_qty"],
+            message_data["product_uom"],
+        )
+        message += "</ul>"
+        return message
+
+    def _prepare_message_data(self, move, request, allocated_qty):
+        return {
+            "request_name": request.name,
+            "picking_name": move.picking_id.name,
+            "product_name": move.product_id.name_get()[0][1],
+            "product_qty": allocated_qty,
+            "product_uom": move.product_uom.name,
+            "location_name": move.location_dest_id.name_get()[0][1],
+        }
+
     def _action_done(self, cancel_backorder=False):
+        for move in self.filtered(lambda m: m.exists() and m.allocation_ids):
```

Comment:
@adrienpeiffer Is this treatment could not be done in check_done() instead ?

Reply:
I'm not sure I understand. Do you mean in the check_done method of stock request? If so, then my answer is no. The check_done method is responsible for checking if we can change the status of the stock request not for changing the quantities on the allocations.

Reply:
Ok, so, moving that code in a separate function should be better.

Reply:
Which part of the code do you mean ?